### PR TITLE
Fix compile warnings and memory leaks

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

**Comment by jschmidb** [comment]
lgtm

**Review by jschmidb — APPROVED** [review]

**Comment by ifranzki** [comment]
Added one more commit, please re-review.

File: src/Makefile.am [hunk 1]
```diff
@@ -90,8 +90,6 @@ icastats_SOURCES = icastats.c icastats_shared.c include/icastats.h
 # internal tests
 
 if ICA_INTERNAL_TESTS
-bin_PROGRAMS += internal_tests/ec_internal_test
-
```
Comment (jschmidb) [inline src/Makefile.am:94]: Good catch! Yes, we don't want the internal tests to be installed. 
However, when I test this change, I'm getting:
```
# ./bootstrap.sh
...
src/Makefile.am:103: warning: variable 'internal_tests_ec_internal_test_SOURCES' is defined but no program or
src/Makefile.am:103: library has 'internal_tests_ec_internal_test' as canonical name (possible typo)
src/Makefile.am:102: warning: variable 'internal_tests_ec_internal_test_LDADD' is defined but no program or
src/Makefile.am:102: library has 'internal_tests_ec_internal_test' as canonical name (possible typo)
```
Besides of this warning, everything seems to be fine and make install does no longer install the internal tests. I googled for a solution of this warning and indeed found some hints, but nothing really worked here.

Comment (ifranzki) [reply src/Makefile.am:94]: Arg, didn't see that. And Travis is also failing. 
Let me look, this needs to be fixed.....

Comment (ifranzki) [reply src/Makefile.am:94]: noinst_PROGRAMS is what we need. Will push shortly....

**Comment by ifranzki** [comment]
Should be OK now.

**Comment by jschmidb** [comment]
yep, runs perfect now.

**Pull request merged**